test/functional/eval/system_spec.lua

```diff
@@ -352,8 +371,16 @@ describe('systemlist()', function()
       eq(1, eval('v:shell_error'))
       eval([[systemlist("cmd.exe /c exit 5")]])
       eq(5, eval('v:shell_error'))
+      eq('root = true\r', eval([[get(systemlist('"more" ".editorconfig"'), 0, '')]]))
+      eq(0, eval('v:shell_error'))
       eval([[systemlist('this-should-not-exist')]])
       eq(1, eval('v:shell_error'))
+      eval([[systemlist('"ping" "-n" "1" "127.0.0.1"')]])
+      eq(0, eval('v:shell_error'))
+      eq('"a b"\r', eval([[get(systemlist('cmd /s/c "cmd /s/c "cmd /s/c "echo "a b""""'), 0, '')]]))
+      eq(0, eval('v:shell_error'))
+      eq('"a b"\r', eval([[get(systemlist('powershell -NoProfile -NoLogo -ExecutionPolicy RemoteSigned -Command echo ''\^"a b\^"'''), 0, '')]]))
+      eq(0, eval('v:shell_error'))
```

Comment:
These new tests aren't relevant to this `sets v:shell_error` test group. 

If we really need to check v:shell_error for these quote/escape tests, just do it in the other tests.  I don't think we do though.  

We need to keep the tests readable where possible.